File: py-abi/examples/track_children.py
import sys
import time
from py_query_system_info import start_tracking_children

if len(sys.argv) < 2:
    print("Usage: python track_children.py <pid>")
    sys.exit(1)

pid = int(sys.argv[1])
print(f"Tracking children of PID: {pid}")

def on_child_process(child):
    print("\n[New Child Process]")
    print(f"  PID: {child['pid']}")
    print(f"  PPID: {child['ppid']}")
    print(f"  Name: {child['name']}")
    print(f"  Command: {' '.join(child['cmdline'])}")
    print(f"  Exe: {child['exe_path']}")

tracker = start_tracking_children(pid, on_child_process)
print("Tracking started. Press Ctrl+C to stop...\n")

try:
    time.sleep(60)
finally:
    tracker.stop()
    print("\nTracking stopped.")

File: examples/track_children.rs
use query_system_info::process;
use std::thread;
use std::time::Duration;

fn main() {
    let pid = std::env::args()
        .nth(1)
        .and_then(|s| s.parse::<u32>().ok())
        .unwrap_or_else(|| {
            eprintln!("Usage: cargo run --example track_children <pid>");
            std::process::exit(1);
        });

    println!("Tracking children of PID: {}", pid);

    let tracker = process::start_tracking_children(pid, |child| {
        println!("\n[New Child Process Detected]");
        println!("  PID: {}", child.pid);
        println!("  PPID: {}", child.ppid);
        println!("  Name: {}", child.name);
        println!("  Command: {}", child.cmdline.join(" "));
        println!("  Exe: {}", child.exe_path);
    })
    .expect("Failed to start tracking");

    println!("Tracking started. Press Ctrl+C to stop...\n");

    thread::sleep(Duration::from_secs(60));

    tracker.stop();
    println!("\nTracking stopped.");
}
